### ISSUE TITLE
Error Unauthorized when using TryItOut Executor

### ISSUE THREAD

**Issue by meelash** [swagger-api/swagger-js#1476]
### Q&A (please complete the following information)
 - OS: macOS
 - Environment: Node.js v. 12.6.0
 - Method of installation: npm
 - Swagger-Client version: 3.9.4
 - Swagger/OpenAPI version: Swagger 2.0

Swagger/OpenAPI definition:
```json

    "/v2/tables/{tableName}/records": {
      "get": {
        "tags": [
          "Tables"
        ],
        "summary": "Returns records from the table",
        "operationId": "TablesV2_GetRecords",
        "consumes": [
          "application/json"
        ],
        "produces": [
          "application/json",
          "text/json",
          "application/xml",
          "text/xml"
        ],
        "parameters": [
          {
            "name": "tableName",
            "in": "path",
            "description": "Table name",
            "required": true,
            "type": "string"
          },
          {
            "name": "q.select",
            "in": "query",
            "description": "List of fields separated by comma",
            "required": false,
            "type": "string"
          },
          {
            "name": "q.where",
            "in": "query",
            "description": "WHERE clause",
            "required": false,
            "type": "string"
          },
          {
            "name": "q.groupBy",
            "in": "query",
            "description": "GROUP BY clause",
            "required": false,
            "type": "string"
          },
          {
            "name": "q.orderBy",
            "in": "query",
            "description": "ORDER BY clause",
            "required": false,
            "type": "string"
          },
          {
            "name": "q.limit",
            "in": "query",
            "description": "Maximum number of records returned (skipped when q.pageNumber or q.pageSize is not empty, max = 1000, default = 100)",
            "required": false,
            "type": "integer",
            "format": "int32"
          },
          {
            "name": "q.pageNumber",
            "in": "query",
            "description": "Page number (if q.pageSize is not empty then default = 1)",
            "required": false,
            "type": "integer",
            "format": "int32"
          },
          {
            "name": "q.pageSize",
            "in": "query",
            "description": "Number of records per page (possible values from 5 to 1000, if q.pageNumber is not empty then default = 25)",
            "required": false,
            "type": "integer",
            "format": "int32"
          }
        ],
        "responses": {
          "200": {
            "description": "OK",
            "schema": {
              "$ref": "#/definitions/GetRecordsResponse"
            }
          }
        }
      },
  "securityDefinitions": {
    "Bearer": {
      "type": "apiKey",
      "description": "Enter the bearer token below. If you don’t have one yet, generate your token following the instructions on Caspio’s help article: <a target=\"_blank\" href=\"http://howto.caspio.com/?p=15781\">Authenticating REST API.</a>",
      "name": "Authorization",
      "in": "header"
    }
  }
```

### How can we help?
I'm having trouble figuring out exactly what to do for authorization.
The relevant part of schema is above. That is all that is defined with regards to security in the schema and one sample endpoint.

I am creating an instance using the constructor as follows:
```
swaggerClient = await Swagger(caspioApiSpecEndpoint, {
    requestInterceptor: (req)=> {
      req.headers['Authorization'] = 'Bearer ' + token
    }
  });
```

Using this, I am able to access the api using the tags interface, for example:
```
swaggerClient.apis.Tables.TablesV2_GetRecords({
      tableName: someTable,
      "q.select": "Id, name, address",
      "q.limit": 1000
})
```
I would like to do this, instead:
```
swaggerClient.execute({
    pathName: '/v2/tables/{tableName}/records',
    method: 'get',
    parameters: {
        tableName: someTable,
        "q.select": "Id, name, address",
        "q.limit": 1000
    }
})
```

This fails with an error 'Unauthorized'. It also fails if I set up the instance as:
```
swaggerClient = await Swagger(caspioApiSpecEndpoint, {
    authorizations: {
        Bearer: token
  });
```

Same error. I've tried lots of other combinations of Bearer, Authorization, "Bearer " +token, etc. Nothing seems to work. What am I missing?

(a) Why can't the tryitout executor work with the injected header, and what is the correct format for the securities definition on an instance wide level, as opposed to request by request.

**Comment by meelash** [comment]
I've tried:
```
swaggerClient = await Swagger(caspioApiSpecEndpoint, {
    authorizations: {
      Authorization: 'Bearer ' + token
    }
  });
```

```
swaggerClient = await Swagger(caspioApiSpecEndpoint, {
    authorizations: {
      Authorization: token
    }
  });
```

```
swaggerClient = await Swagger(caspioApiSpecEndpoint, {
    authorizations: {
      Bearer: 'Bearer ' + token
    }
  });
```

etc...

**Comment by meelash** [comment]
Also, what does "TryItOut" mean? What is this referring to? I feel like maybe I'm missing something big here, in terms of what this is supposed to do... and is there not supposed to be a buildRequest instance method, the way there is a static method?

**Comment by markopy** [comment]
I had the same issue with `requestInterceptor` passed when creating the client not being used with `client.execute()`. 
This might be intentional given that it's called the "TryItOut Executor" but the readme in the github repo is very confusing about all of this. If the tags interface is the primary way this library is supposed to be used (which I would say it is) then that should appear first in the documentation and the `http()` fetch api and tryout executor should be at the bottom as ancillary.

**Comment by char0n** [comment]
@meelash "TryItOut" refers to "Try it out" functionality in [swagger-ui](https://github.com/swagger-api/swagger-ui). If you open this [swagger-ui demo](https://petstore.swagger.io/#/pet/uploadFile) you can see the button with label `Try it out`. This button is using `swagger-client` (this library) when you click on it under the hood.  `request building` + `execution` part of `swagger-client` is generally referred to as `TryItOut` functionality.

Regarding `authorization`; You can find more about how to do authorization in this [TryItOut documentation](https://github.com/swagger-api/swagger-js/blob/master/docs/usage/try-it-out-executor.md).

**Comment by char0n** [comment]
Regarding ability to pass `requestInterceptor` to `TryItOut Executor`; `requestInterceptor` is not a supported property. Refer to this [documentation](https://github.com/swagger-api/swagger-js/blob/master/docs/usage/try-it-out-executor.md) to see the list of all supported properties.

Passing `requestInterceptor` is part of low level API of [HTTP Client](https://github.com/swagger-api/swagger-js/blob/master/docs/usage/http-client.md). 

Technically it would be possible to use `requestInterceptor` even in `TryItOut` usecase. We can use `http` property to achieve that. 

```js
SwaggerClient.execute({
  spec: pojoDefinition,
  operationId: 'getUserList',
  parameters: { q: 'search string' },
  securities: { authorized: { BearerAuth: "3492342948239482398" } },
});

**Comment by markopy** [comment]
@char0n What is confusing is not using `SwaggerClient.execute()` directly but instantiating a client and then calling `.execute()` on it:
```
client = Swagger(spec, {
    requestInterceptor: (req)=> {
      req.headers['someheader'] = 'value'
    }
});
```
with this
```
 client.apis.path.operation(params)
```
will use the spec and requestInterceptor passed when instantiating the client while
```
 client.execute({operationId: op, parameters: params })
```
will use the spec but not requestInterceptor. I would expect the tag api and `execute()` to do the same thing when called on the same client instance.

**Comment by char0n** [comment]
Deleted my last comment as my assumptions there are not true and may be confusing for future reference. 

You're right that `requestInterceptor` is part of `SwaggerClient` API and can  be passed during instantiation as documented [here](https://github.com/swagger-api/swagger-js/blob/master/docs/usage/tags-interface.md).

`execute` and `resolve` instance methods has been introduced as pure convenience to pass the configuration that have been provided via `SwaggerClient` constructor. As the time went by, the `SwaggerClient` constructor  supported more and more configuration options, but those options have not been translated into these instance functions. I don't see a reason why not fix this behavior now, except the fact that by fixing it we'll be changing the expected behavior of those functions which can be considered a breaking change. Let me think about it for a while.

**Comment by char0n** [comment]
After some thought process I decided to issue a PR fixing this behavior. This is a backward compatible change as consumer of the library has an ability to override via `execute` options argument.

**Comment by char0n** [comment]
Will be released in swagger-client@3.10.8

**Comment by meelash** [comment]
I can confirm that this works in the latest swagger-client. Thanks!
